src/react.js

```diff
@@ -3,59 +3,41 @@ import PropTypes from 'prop-types';
 import './polyfill';
 import { importWithDependencies, importDependenciesOf } from './index';
 
-class ExternalComponent extends Component {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      loaded: false,
-    };
-    this.importPromise = this.importPromise.bind(this);
-    this.Component = null;
-  }
-
-  importPromise(src) {
-    if (!src) {
-      return new Promise((resolve, reject) => {
-        reject();
-      });
-    }
-    if (this.props.cors) {
-      return require('./corsImport').default(src);
-    }
-
-    return new Promise((resolve) => {
-      resolve(new Function(`return import("${src}")`)());
-    });
-  }
-
-  componentDidMount() {
+const ExternalComponent = (props) => {
+  const { src, module, export: exportName, coors, ...rest } = props;
+  let Component = null;
+  const [loaded, setLoaded] = useState(false);
+  
+  useEffect(() => {
```

Comment:
`useEffect(()=>{},[])`

without that empty array on the end, `useEffect` behaves like `componentDidMount, and componentDidUpdate` causing multiple executions of the hook